### PR TITLE
fix(plugins): allow for relaxed plugin loading

### PR DESCRIPTION
**Pull request by link108**
These changes were meant to be part of https://github.com/spinnaker/kork/pull/564, I had some local commits I forgot to push 😬

### CODE REVIEW

**Comment by jonsie** [comment]
@link108 Any reason not to just add this functionality to the `SpringPluginStatusProvider`?

**Comment by link108** [comment]
@jonsie I took a look at putting it there before, that class seems like its focused on specific plugin properties vs properties under the `spinnaker.extensibility` key, where the strict loading config goes. Can move it there, but want to be careful about overloading that class

File: kork-plugins/src/main/kotlin/com/netflix/spinnaker/kork/plugins/SpringStrictPluginLoaderStatusProvider.kt [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+/**
+ * Backs strict plugin loading by the Spring environment, instead of using text files.
+ */
+class SpringStrictPluginLoaderStatusProvider(
+  private val environment: Environment
+) {
+
+  fun isStrictPluginLoading(): Boolean =
+    environment.getProperty("spinnaker.extensibility.strict-plugin-loading")?.toBoolean() ?: false
```
Comment (jonsie) [inline kork-plugins/src/main/kotlin/com/netflix/spinnaker/kork/plugins/SpringStrictPluginLoaderStatusProvider.kt]: This should default to true right?

**Comment by jonsie** [comment]
@link108 Ah yea, that makes sense.  I wonder if we should just pass in a boolean to these classes  rather than making it a dedicated class?  If we need to add more configs for this type of "strict" loading stuff we could make it a dedicated class.

**Comment by link108** [comment]
@jonsie I played around with passing a boolean instead of the class and it made testing a bit harder, eg since fixtures are used in testing, we'd be passing a boolean to the constructors, which I found hard to override for specific tests.

**Comment by jonsie** [comment]
@link108 Sounds good, LGTM

**Pull request merged**